### ISSUE TITLE
[FEATURE]: avoid double negation by introducing `enabled:` for modules

### ISSUE THREAD

**Issue by stefreak** [garden-io/garden#3221]
## Feature Request
Deprecate `disabled:` module option and introduce `enabled`.

Alternatively, encourage using `enabled` and still allow `disabled` in the cases where it makes the code more readable.

### Background / Motivation
Starting out positive, both the positive and the negative case are easy to understand:

* `enabled: ${environment.name != "local"}` (read: enabled if environment is not local)
* `enabled: ${environment.name == "local"}` (read: enabled if environment is local)

While starting out negative, the negative case can be harder to read:

* `disabled: ${environment.name != "local"}` (read: disabled if environment is not local)

**Examples with simple boolean algebra**:

Easy to understand:

* `enabled: ${environment.name == "kubernetes" || environment.name == "local-kubernetes"}` (read: the module is enabled if the environment is kubernetes or local-kubernetes)
* `enabled: ${environment.name != "kubernetes" && environment.name != "local-kubernetes"}` (read: the module is enabled if the environment is not kubernetes and not local-kubernetes)
* `disabled: ${environment.name == "kubernetes" || environment.name == "local-kubernetes"}` (read: the module is disabled, if environment is kubernetes or local-kubernetes)

Hard to understand

* `disabled: ${environment.name != "kubernetes" && environment.name != "local-kubernetes"}` (read: the module is disabled, if environment is not kubernetes and is not local-kubernetes)

So, to sum it up, enabled is okay to understand in all cases, and disabled is hard to understand in case of double negation (at least for me, but [also to other people](https://maximilianocontieri.com/code-smell-51-double-negatives)).

### What should the user be able to do?
use `enabled:` to configure wether a module is active, instead of `disabled`.

### Why do they want to do this? What problem does it solve?
Readability

### Suggested Implementation(s)
There are 2 options:

* We can have both `enabled` and `disabled` side by side
* We can deprecate `disabled`

### How important is this feature for you/your team?

:rose: It’s a nice to have, but nice things are nice :slightly_smiling_face:

**Comment by vvagaytsev** [comment]
Thanks, @stefreak! This would be a really nice feature :) We'll take a look soon. Feel free to submit a PR, I can help you if you need any assistance.

**Comment by stefreak** [comment]
@edvald @vvagaytsev @Orzelius any opinions on this? Could we add this to our list of changes for the new schema introduced in 0.13?

**Comment by edvald** [comment]
I'm okay with this. Although I'd actually be in favor of allowing either, but mutually exclusive (i.e. you can either specify neither or exactly one of those fields). In my mind there are about as many cases where you conditionally want to _disable_ something as to _enable_ it .)

**Comment by edvald** [comment]
Please note that changes along these lines could conflict quite aggressively with the 0.13 branch... I'd suggest avoiding until that's merged or doing on that branch.

**Comment by stefreak** [comment]
I'm okay with keeping both, @vvagaytsev @Orzelius I would just create a pull request towards 0.13 once all the tests pass? What do you think?

**Comment by Orzelius** [comment]
I think most of this would be solved by just naming the complicated expression with a variable
```yml
variables:
  isKubernetes: ${environment.name == "kubernetes" || environment.name == "local-kubernetes"}
disabled: ${!var.isKuberetes}
```

**Comment by Orzelius** [comment]
But Still, enabled is easy to understand, I'm just doubting the gain from adding it vs working on some other issue
